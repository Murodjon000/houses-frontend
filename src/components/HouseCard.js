import React from 'react';
import PropTypes from 'prop-types';

const HouseCard = ({ name, image, price }) => {
  const formatToCurrency = (amount) =>
    '$' + ' ' + amount.toFixed(2).replace(/\d(?=(\d{3})+\.)/g, '$&,'); // eslint-disable-line

  return (
    <div className="shadow rounded infoCarousel__wrapper-desc-info mx-auto ">
      <img src={image} className="w-100 border-card" alt="..." />
      <div className="d-flex justify-content-between align-items-center p-3">
        <div>
          <h3 className="text-center font-weight-bold">{name}</h3>
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star text-sm-gray" />
        </div>
        <div>
          <p className="font-weight-bold text-center">
            {formatToCurrency(price)}
          </p>
          <p className="text-sm-gray text-center">Per month</p>
        </div>
      </div>
    </div>
  );
};

HouseCard.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
  price: PropTypes.number,
};

HouseCard.defaultProps = {
  name: '',
  image: '',
  price: 0,
};

export default HouseCard;
