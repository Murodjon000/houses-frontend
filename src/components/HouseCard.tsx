import React from 'react';
import PropTypes from 'prop-types';
import { Link } from '@reach/router';
import formatToCurrency from '../helpers/currency_format';

// eslint-disable-next-line
const HouseCard = ({ name, image, price, id }) => (
  <Link to={`/houses/${id}`}>
    <div className="shadow rounded infoCarousel__wrapper-desc-info mx-auto ">
      <img src={image} className="w-100 border-card" alt="..." />
      <div className="d-flex bg-white justify-content-between align-items-center p-3">
        <div>
          <h3 className="text-center font-weight-bold test-class">{name}</h3>
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star checked" />
          <span className="fa fa-star text-sm-gray" />
        </div>
        <div>
          <p className="font-weight-bold text-center">
            {formatToCurrency(price)}
          </p>
          <p className="text-sm-gray text-center">Per month</p>
        </div>
      </div>
    </div>
  </Link>
);

HouseCard.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
  price: PropTypes.number,
  id: PropTypes.string,
};

HouseCard.defaultProps = {
  name: '',
  image: '',
  price: 0,
  id: '',
};

export default HouseCard;
