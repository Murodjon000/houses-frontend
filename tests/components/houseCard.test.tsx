import * as React from 'react';
import * as renderer from 'react-test-renderer';
import HouseCard from '../../src/components/HouseCard';

test('Link changes the class when hovered', () => {
  const price = 5000;
  const component = renderer.create(
    <HouseCard
      name="Family house"
      price={price}
      id="2"
      image="https://images.unsplash.com/photo-1603993097397-89c963e325c7?ixid=MnwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=750&q=80"
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
