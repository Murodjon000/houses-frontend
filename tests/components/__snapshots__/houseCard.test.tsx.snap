// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  href="/houses/2"
  onClick={[Function]}
>
  <div
    className="shadow rounded infoCarousel__wrapper-desc-info mx-auto "
  >
    <img
      alt="..."
      className="w-100 border-card"
      src="https://images.unsplash.com/photo-1603993097397-89c963e325c7?ixid=MnwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=750&q=80"
    />
    <div
      className="d-flex bg-white justify-content-between align-items-center p-3"
    >
      <div>
        <h3
          className="text-center font-weight-bold test-class"
        >
          Family house
        </h3>
        <span
          className="fa fa-star checked"
        />
        <span
          className="fa fa-star checked"
        />
        <span
          className="fa fa-star checked"
        />
        <span
          className="fa fa-star checked"
        />
        <span
          className="fa fa-star text-sm-gray"
        />
      </div>
      <div>
        <p
          className="font-weight-bold text-center"
        >
          $ 5,000.00
        </p>
        <p
          className="text-sm-gray text-center"
        >
          Per month
        </p>
      </div>
    </div>
  </div>
</a>
`;
